mockgcp/mock_http_roundtrip.go

```diff
@@ -94,7 +96,23 @@ type MockService interface {
 	ExpectedHost() string
 }
 
-func NewMockRoundTripper(t *testing.T, k8sClient client.Client, storage storage.Storage) *mockRoundTripper {
+type Interface interface {
+	// We support HTTP requests
+	http.RoundTripper
+
+	// NewGRPCConnection returns a grpc connection to our mock implementation
+	NewGRPCConnection(ctx context.Context) *grpc.ClientConn
+
+	// We can dispatch test commands
+	SupportsTestCommands
+}
+
+type SupportsTestCommands interface {
+	// RunTestCommand is a "backdoor" into our mock implementation that is useful for fault injection or faking scaling events etc
+	RunTestCommand(ctx context.Context, service string, command string) error
```

Comment:
Suggest calling out that the `RunTestCommand` pairs with 'MockGCPBackdoor' env var (unless we really want to keep this feature silent 🤫 )

Reply:
Good call, I'll add it (in a follow-on PR)